text/0000-module-resolution.md

```diff
@@ -0,0 +1,356 @@
+
+### LWC configuration
+
+A configuration file will be located at the root of the project, optionally at the component namespace level and in the future (this is yet to be spec out) at the module bundle level, where specificity will be resolved from inner to outer (merging upwards - similar to node_modules).
+
+At the package root level there are two ways on which you can add a configuration file: Creating a `lwc.config.json` or adding an `lwc` key to package.json (this is mostly for simplicity and easy of use - specially when publishing a package). Here are some examples:
+
+```json5
+// package.json
+{
+    "name: "foo",
+    "version": "bar",
+    "lwc": {
+        "modules": [
+            /* list with modules records goes here */
+        ]
+    }
+}
+```
+
+```json5
+// lwc.config.json
+{
+    "modules:" [
+        /* list with modules records goes here */
+    ]
+}
+```
+
+Its important that **any plugin, tool or project honor this configuration**, otherwise the resolution will not work correctly in most environments.
+
+Moreover, note that this type of configuration is not something new, many tools use an equivalent configuration (babel, lerna, rollup, webpack, ...)
+
+### Module resolution configuration
+
+The `modules` key on the configuration will hold an Array of `ModuleRecords`. A module record is an object with a specific shape depending on the type of module record it holds. The type will depend on the mode of resolution, we will be defining three types of `ModuleRecords` for now:
+  - AliasModuleRecord: Define an specifier that points to a specific module.
+  - DirectoryModuleRecord: Defines a folder where many namespaced modules can be defined
+  - NpmModuleRecord: Defines a set of packages that will hold more modules to be resolved.
+
+### Why Objects instead of raw strings?
+
+In previous draft versions, `module` entries used to be just *strings* which had slightly better ergonomics. However, raw strings present a lot of ambiguities, let's look at one example:
+
+```json
+{
+    "modules": [
+        "just-string"
+    ]
+}
+```
+
+On the example below, it will be impossible to know if `just-string` is the name of an npm package, or a name of a folder that contains components, hence creationg resolution ambiguity.
+
+We considered other alternatives such as forcing specific prefix like `npm:package-name` or require a final slash (ex. `folder-name/`) for folders, but we believed that those were more cumbersome to learn and future-proof, so standarizing on an consistent object shape as ModuleRecord was a preferable and superior approach.
+
+### Types of ModuleRecord
+
+#### AliasModuleRecord
+
+This type of resolution maps a given module specifier to a particular path within the root of that module.
+
+```json
+{
+    "modules": [
+       {
+            "name": "ui/button",
```

Comment:
I think it should also be stated in this sections that: 
* the `name` property should have the following form `<ns>/<name>` where `ns` is a valid module name and `name` is a valid module name.
* the directory structure doesn't have any baring on the module name and namespace. In the following example, the component namespace will be `ui` and the component name will be `button`.

```json
{
  "modules": [
    {
      "name": "ui/button",
      "path": "src/module/foo/bar/bar.js"
    }
  ]
}
```